spec/features/consumer/shopping/cart_spec.rb

```diff
@@ -57,6 +57,7 @@
 
       before do
         add_product_to_cart order, product_tax
+        Spree::Config.allow_backorders = false
```

Comment:
This could affect other tests since the Spree Config caches settings. A possible fix is discussed in https://github.com/openfoodfoundation/openfoodnetwork/pull/1742. I guess we should at least restore the original value of `allow_backorders` after the test.

Reply:
I think `allow_backorders = false` is the default setting. These changes to the cart spec were because of this (from slack):

"it looks like adding more tests changed the times at which the various specs would be running on the different Travis nodes, into the exact order that one of the specs dealing with "allow_backorders" was running at the same time as a cart spec on a different node, which meant a test to see if the cart showed an out of stock message was running without it raising the expected stock level error... hence why it was working on my local machine but not on Travis..."

Reply:
@Matt-Yorkley as I explained on slack, I don't think your understanding of the issue is quite right here. The problem is not caused by concurrency of travis nodes, but by specs being reordered or shifted to different nodes. By not resetting `Spree::Config` values to whatever they were when the spec started running, we just open up the potential for more issues like the one you experienced in the future.

Reply:
@oeoeaio I get that it was triggered this time by specs shifting onto different nodes, but it's also caused by the concurrency of those shifted specs, right?

I'm pretty sure `allow_backorders = false` is the default setting. If you grep our specs folder there's 7 specs that set `allow_backorders = false` without resetting it (I assume for the same reason), and only one spec that sets it temporarily to `true` and then resets it to `false` immediately after. From previous discussions on `allow_backorders` I think it was decided it should always be set to `false` by default on production, as having it enabled creates lots of problems.

Reply:
It allows adding to cart and ordering of items that are out of stock, across every enterprise in the instance, which breaks lots of specs if set to `true`.

Reply:
@Matt-Yorkley Yep I understand all that, but if you look at where this preference is defined in Spree (`core/app/models/spree/app_configuration.rb`) you will see that it actually defaults to `true`. So I assume that the reason that this particular spec was failing for you is that after it moved to a new node, it was no longer preceded by a spec that set the value to `false` and so it was using the default value of `true`, thus causing the spec to fail. There is no issue with concurrency between nodes, I promise.

It is getting to the point where we need to explicitly set all of these variables in `config/test.rb` so that we actually know what environment our specs are running in. I might look at doing that as part of #1742.

Reply:
These cart specs and a number of others all fail if `allow_backorders ` is enabled. I'm pretty sure all instances have this option disabled, even though it defaults to enabled in Spree. 

I think it's something that should probably be set in `test.rb`, as you say.

Reply:
I've added a discussion here to get more feedback: https://community.openfoodnetwork.org/t/backorders/1080

Reply:
>  I'm pretty sure all instances have this option disabled, even though it defaults to enabled in Spree.

Yep, that is right.